src/shared/pam-util.h

```diff
@@ -9,6 +9,12 @@ int pam_syslog_errno(pam_handle_t *handle, int level, int error, const char *for
 
 int pam_syslog_pam_error(pam_handle_t *handle, int level, int error, const char *format, ...) _printf_(4,5);
 
+/* Call pam_vsyslog if debug is enabled */
+#define pam_debug_syslog(handle, fmt, ...) ({                           \
+                        if (debug)                                      \
+                                pam_syslog(handle, LOG_DEBUG, fmt, ## __VA_ARGS__); \
```

Comment:
I reworked this to avoid using a local variable by fixed name. But I didn't like the `dbghdl` approach because it doesn't work nicely with the way that the functions have no global state and pass both `handle` and `debug` through parameters. Instead, an explicit parameter is passed. This is very clear and obvious and still relatively concise.

Reply:
Another reason why abusing `pam_handle_t*` might not be a good idea is that `pam_syslog` allows `pam_handle_t*` to be NULL.